### PR TITLE
[moveit_cpp] Fix double declaration

### PR DESCRIPTION
**Pull request by galou**
The plugin did throw an error that parameters were already declared, at
least when I used it.
This commit fixes this.

Signed-off-by: Gaël Écorchard <gael.ecorchard@cvut.cz>

### Description

Please explain the changes you made, including a reference to the related issue if applicable

### Checklist
- [x] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_planners/pilz_industrial_motion_planner/include/joint_limits/joint_limits_rosparam.hpp [hunk 1]
```diff
@@ -27,6 +27,16 @@
...
+/** declare a parameter if not already declared. */
+#define DECLARE(node, name, type, default_value)                                                                       \
+  do                                                                                                                   \
+  {                                                                                                                    \
+    if (not node->has_parameter(name))                                                                                 \
+    {                                                                                                                  \
+      node->declare_parameter<type>(name, default_value);                                                              \
+    }                                                                                                                  \
+  } while (false);
+
```
Comment (tylerjw) [inline moveit_planners/pilz_industrial_motion_planner/include/joint_limits/joint_limits_rosparam.hpp:39]: I don't understand the need for the do-while?

Comment (vatanaksoytezer) [reply moveit_planners/pilz_industrial_motion_planner/include/joint_limits/joint_limits_rosparam.hpp:39]: I didn't understand this either.

Comment (galou) [reply moveit_planners/pilz_industrial_motion_planner/include/joint_limits/joint_limits_rosparam.hpp:39]: By habit. Should I remove it?

Comment (vatanaksoytezer) [reply moveit_planners/pilz_industrial_motion_planner/include/joint_limits/joint_limits_rosparam.hpp:39]: Yeah I would prefer just a clean if statement here.

Comment (tylerjw) [reply moveit_planners/pilz_industrial_motion_planner/include/joint_limits/joint_limits_rosparam.hpp:39]: I would generally really prefer a template function over a c-macro.  It should be fairly straightforward to make this a template function.

**Review by tylerjw — APPROVED** [review]
This is a simple and nice change, thank you!

**Comment by tylerjw** [comment]
This isn't passing CI because CI itself is broken during the transition to Jammy.  There is another PR that should fix that soon and I'll rebase and merge that afterward.

**Review by JafarAbdi — APPROVED** [review]

**Review by henningkayser — APPROVED** [review]

**Pull request merged**